stuffer/s2n_stuffer_file.c

```diff
@@ -67,20 +67,27 @@ int s2n_stuffer_send_to_fd(struct s2n_stuffer *stuffer, int wfd, uint32_t len)
 
 int s2n_stuffer_alloc_ro_from_fd(struct s2n_stuffer *stuffer, int rfd)
 {
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
     struct stat st = {0};
 
-    S2N_ERROR_IF(fstat(rfd, &st) < 0, S2N_ERR_FSTAT);
+    ENSURE_POSIX(fstat(rfd, &st) < 0, S2N_ERR_FSTAT);
```

Comment:
```suggestion
    ENSURE_POSIX(fstat(rfd, &st) >= 0, S2N_ERR_FSTAT);
```

Reply:
Can we just `GUARD` in this case?

Reply:
The `s2n_erno` wouldn't be set if we did. It might be valuable to create a `GUARD_AS(expr, code)` and `GUARD_POSIX_AS(expr, code)`.